### DISCUSSION TITLE
公式旋转90度

### DISCUSSION THREAD

**Discussion by qiaoxiaoyue** [sjtug/SJTUThesis#779] [提问] [answered]
因为矩阵太大，导致公式过长，需要将公式旋转90度，在页面中横着展示，在网上没找到可用的方法，特此请教，希望有经验的同学可以帮帮忙。

**Comment by LogCreative** [comment]
可以试试 `lscape` 宏包的 `landscape` 环境让整个页面横置
```latex
% 导言区...
% \usepackage{lscape}
% \usepackage{physics} % 仅为了方便输入矩阵
\begin{landscape}     % 横向环境，注意页眉没有旋转
    \begin{figure}[p]    % 浮动体专页，为了纵向居中
    \begin{equation}  
        \mathbf{A} = \begin{pmatrix}
            \xmat*{a}{9}{9}  % 需要 physics
        \end{pmatrix}
    \end{equation}
    \end{figure}
\end{landscape}
```

<img width="387" alt="image" src="https://user-images.githubusercontent.com/61653082/161994532-f25da4d3-60a2-4372-b60c-2d9d04913669.png">

**Comment by AlexaraWu** [comment, marked as answer]
也可以试试模板自带 `graphicx` 包提供的 `\rotatebox`

```latex
\begin{equation}
  \rotatebox[origin=c]{90}{$
    \displaystyle
    \mathbf{A} = \begin{pmatrix}
      \xmat*{a}{9}{9}  % 需要 physics
    \end{pmatrix}
  $}
\end{equation}
```

在 `\rotatebox` 的参数中输入旋转的内容时，需要加 `$` 重新进入数学环境，并使用 `\displaystyle` 切换为行间公式的样式。

![w](https://user-images.githubusercontent.com/15205102/162120332-cc56b2ac-1e0a-4271-8ed3-5a4f07ef28d0.png)
